hive-bigquery-connector-common/src/main/java/com/google/cloud/hive/bigquery/connector/utils/hive/HiveUtils.java

```diff
@@ -73,9 +85,38 @@ private static JobID getJobIDWithVertexAppended(JobID jobID, String vertexId) {
     }
   }
 
-  private static class TaskAttemptIDWrapper extends TaskAttemptID {
+  public static String getTaskID(Configuration conf) {
+    if (isSparkJob(conf)) {
+      return getSparkTaskID();
+    }
+    return getHiveTaskAttemptIDWrapper(conf).getTaskID().toString();
+  }
+
+  /**
+   * Retrieves the Spark task ID from the Spark context. Uses reflection to avoid this library
+   * requiring dependencies on Spark packages.
+   */
+  public static String getSparkTaskID() {
+    try {
+      Class<?> taskContextClass = Class.forName("org.apache.spark.TaskContext");
+      Method getMethod = taskContextClass.getMethod("get");
+      Object taskContext = getMethod.invoke(null);
+      Method partitionIdMethod = taskContextClass.getMethod("partitionId");
+      Object partitionId = partitionIdMethod.invoke(taskContext);
+      Method stageIdMethod = taskContextClass.getMethod("stageId");
+      Object stageId = stageIdMethod.invoke(taskContext);
+      return String.format("stage-%s-partition-%s", stageId, partitionId);
```

Comment:
right now all stream ref files are dumped into a parent path of .../<table>/ and we are assuming  there is only 1 stage write to the table? if there are multiple stages write into same table, then for each stage all the stream ref files will be picked up, this will have data correctness issue.

Reply:
I've updated the code to use the ".hive-staging-XXXX" folder name, when present.

Reply:
See the `getQueryTempOutputPath()` method. We check for the presence of the `FileOutputFormat.OUTDIR` (`mapreduce.output.fileoutputformat.outputdir`) property, which is set by Spark. If present, then we extract the "hive-staging-XXXX" part of that folder name and then use that for our own temp output dir.